selfdrive/loggerd/tests/test_loggerd.py

```diff
@@ -107,25 +104,41 @@ def test_init_data_values(self):
     for _, k, v in fake_params:
       self.assertEqual(getattr(initData, k), v)
 
-  # TODO: this shouldn't need camerad
-  @with_processes(['camerad'])
   def test_rotation(self):
     os.environ["LOGGERD_TEST"] = "1"
     Params().put("RecordFront", "1")
+
     expected_files = {"rlog.bz2", "qlog.bz2", "qcamera.ts", "fcamera.hevc", "dcamera.hevc"}
+    streams = [(VisionStreamType.VISION_STREAM_ROAD, tici_f_frame_size if TICI else eon_f_frame_size, "roadCameraState"),
+              (VisionStreamType.VISION_STREAM_DRIVER, tici_d_frame_size if TICI else eon_d_frame_size, "driverCameraState")]
     if TICI:
       expected_files.add("ecamera.hevc")
+      streams.append((VisionStreamType.VISION_STREAM_WIDE_ROAD, tici_e_frame_size, "wideRoadCameraState"))
 
-    # give camerad time to start
-    time.sleep(3)
+    pm = messaging.PubMaster(["roadCameraState", "driverCameraState", "wideRoadCameraState"])
+    vipc_server = VisionIpcServer("camerad")
+    for stream_type, frame_size, _ in streams:
+      vipc_server.create_buffers(stream_type, 40, False, *(frame_size))
+    vipc_server.start_listener()
 
     for _ in range(5):
       num_segs = random.randint(2, 5)
       length = random.randint(1, 3)
       os.environ["LOGGERD_SEGMENT_LENGTH"] = str(length)
-
       managed_processes["loggerd"].start()
-      time.sleep(num_segs*length + 1)
+
+      fps = 20.0
+      for n in range(1, int(num_segs*length*fps)+1):
```

Comment:
we should make a nice helper for sending fake frames and messages. something like this:
```
with FakeMessages(services=['carState', 'deviceState']) as fm, FakeFrames() as ff:
  # start loggerd
  time.sleep(10)
```